### ISSUE TITLE
`c8y_configuration_plugin` doesn't receive new messages after `mosquitto` restarts

### ISSUE THREAD

**Issue by rina23q** [thin-edge/thin-edge.io#1287]
**Describe the bug**
c8y_configuration_plugin doesn't receive new MQTT messages after mosquitto restarts during the runtime of c8y_configuration_plugin.

**To Reproduce**
This steps are copied from [here](https://github.com/thin-edge/thin-edge.io/issues/1184#issuecomment-1185537298).

**[Edited on 2nd of September] Updated the step1 and step2 as we changed the topic to listen.**
[Precondition]
- All thin-edge packages are installed in the device.
- The device is already connected to c8y by `sudo tedge connect c8y`.
- The c8y_configuration_plugin process is up. Run the binary directly (execute `c8y_configuration_plugin` in the shell). 

[Steps]
1. Publish a configuration file change request MQTT message.
```
tedge mqtt pub 'c8y/s/ds' '524,DeviceSerial,http://www.my.url,type'
```
2. Check the output of `c8y_configuration_plugin`. You can see that the plugin received the message by 1).
```
2022-09-02T10:45:48.284185396Z ERROR c8y_configuration_plugin::download: The configuration download for 'type' failed.
2022-09-02T10:45:48.284497529Z ERROR c8y_configuration_plugin: Handling of operation: '524,DeviceSerial,http://www.my.url,type' failed with The requested config_type type is not defined in the plugin configuration file.
```
3. Stop `mosquitto` service.
```
sudo systemctl stop mosquitto
```
4. Start `mosquitto` service.
```
sudo systemctl start mosquitto
```
5. Publish the same message as 1).
```
tedge mqtt pub 'tedge/configuration_change/c8y-configuration-plugin' ''
```
6. Check the output of `c8y_configuration_plugin`. You can't see that the plugin received the new message by 5). 

**Expected behavior**
After restarting `mosquitto`, still `c8y_configuration_plugin` can receive MQTT messages published by anywhere onto the topics `c8y/s/ds`.

**Environment (please complete the following information):**
- OS [Ubuntu 20.04]
- Hardware [Laptop]
- System-Architecture [amd64]
- thin-edge.io version [0.7.2]

**Additional context**
#1286 is the same bug occurred in `c8y_log_plugin`. Since the both plugins have quite similar implementation for MQTT clients, the fix will be the same.

`tedge_mapper c8y` is actually using the same MQTT crates, however, it doesn't have any issue after restarting `mosquitto`. I think the issue is mqtt client implementation of `c8y_configuration_plugin`.

**Comment by PradeepKiruvale** [comment]
steps to test this

Install the thin-edge from here: https://github.com/thin-edge/thin-edge.io/actions/runs/3015303111
1. subscribe to `c8y/s/us` using `sudo tedge mqtt sub 'c8y/s/us'`
2. `sudo systemctl start c8y-log-plugin.service`
3. Wait for some time
4. `sudo tedge connect c8y` 
5. The subscriber that started in the step 1, should  receive `118,software-management`

**Comment by PradeepKiruvale** [comment]
resolved with #1360

**Comment by gligorisaev** [comment]
Tested:
`pi@raspberrypi:~ $ sudo tedge mqtt sub 'c8y/s/us'
INFO: Connected

ERROR: Mqtt state: Io error Custom { kind: ConnectionAborted, error: "connection closed by peer" }
INFO: Connected
[c8y/s/us] 114,c8y_Restart,c8y_LogfileRequest,c8y_SoftwareUpdate,c8y_UploadConfigFile,c8y_DownloadConfigFile

[c8y/s/us] 500

[c8y/s/us] 118,
[c8y/s/us] 118,software-management`

`pi@raspberrypi:~ $ sudo tedge mqtt sub 'c8y/s/us'
INFO: Connected
ERROR: Mqtt state: Io error Custom { kind: ConnectionAborted, error: "connection closed by peer" }
INFO: Connected
[c8y/s/us] 114,c8y_Restart,c8y_LogfileRequest,c8y_SoftwareUpdate,c8y_UploadConfigFile,c8y_DownloadConfigFile

[c8y/s/us] 500

[c8y/s/us] 119,c8y-configuration-plugin`
